### PR TITLE
Push res over websockets

### PR DESCRIPTION
**Pull request by aleksanb**

### CODE REVIEW

**Comment by aleksanb** [comment]
Regex shorthand for or! (|)

File: nin/backend/serve.js [hunk 1]
```diff
@@ -39,19 +39,65 @@ var serve = function(projectPath, shouldRunHeadlessly) {
...
     var sockets = express();
     var sockets_server = require('http').createServer(sockets);
     var sock = socket(projectPath, function(conn) {
       for (var i in watcher.paths) {
-        conn.send('add', {
-          path: watcher.paths[i]
-        });
-      }
+        conn.send('add', eventFromPath({path: watcher.paths[i]}));
+      };
```
Comment (stianjensen) [inline nin/backend/serve.js]: No semicolon needed here, I guess

**Comment by aleksanb** [comment]
Ping. Loder / Loader is also rebased in, just github that's still showing @sigvef s comment.

**Comment by stianjensen** [comment]
The commit doesn't seem to be rebased in. @sigvef 's comment should disappear completely when you've rebased that commit

**Comment by aleksanb** [comment]
Ping. The change is now in the correct commit, i had rebased the fix into the newest commit, not the oldest.

**Comment by cristeahub** [comment]
This needs rebase now!

**Comment by aleksanb** [comment]
PING. Cleaned up some of the code also.

**Comment by aleksanb** [comment]
Turns out the bug was poor file watching performance on linux.
Added fix for that.

**Comment by aleksanb** [comment]
PING. Finalized this feature now.

File: nin/backend/serve.js [hunk 2]
```diff
@@ -39,19 +39,61 @@ var serve = function(projectPath, shouldRunHeadlessly) {
...
+        }
+      });
+      return shaderNames;
+    };
+
+    var eventFromPath = function(data) {
+      var path = data.path,
+          filename = p.basename(path),
+          filenameWithoutExtension = filename.split('.')[0],
+          content = fs.readFileSync(path, 'utf-8');
```
Comment (cristeahub) [inline nin/backend/serve.js]: This is an error just waiting to happen, lol

**Comment by cristeahub** [comment]
Just to be sure before I merge this, as there's no description here:

This is so that we push resources over websockets instead of fetching them over http from the frontend?

**Comment by stianjensen** [comment]
> This is so that we push resources over websockets instead of fetching them over http from the frontend?

Yup. The main feature though, is that you don't need to manually declare dependencies on shaders anymore, because the backend searches for references to `SHADERS.{whatever}` and generates a dependency list automatically

File: nin/backend/serve.js (hunk 2 above)
Comment (aleksanb) [inline nin/backend/serve.js]: It's so visually unpleasing with all the var's though!
We can add a linter to the project though that can enforce whatever we might decide on regarding declarations.

**Comment by cristeahub** [comment]
LGTM :+1:

**Pull request merged**